openedx/features/announcements/views.py

```diff
@@ -80,3 +82,31 @@ def get(self, request, **kwargs):
         if not created:
             announcement.delete()
         return HttpResponseRedirect(reverse('announcements:announcements_list'))
+
+class AnnouncementsJSONView(View):
+    """
+    View returning a page of announcements for the dashboard
+    """
+
+    def get(self, request, **kwargs):
+        """
+        Return active announcements as json
+        """
+        page = kwargs.get("page", 1)
+        announcements = Announcement.objects.filter(active=True)
+        paginator = Paginator(announcements, 5)
```

Comment:
`5` should be configurable.

Reply:
Ah, good idea. I've added a setting for this under settings.ANNOUNCEMENT_SETTINGS.